### PR TITLE
deps: bump bdk to beta_5

### PR DESCRIPTION
**Pull request by reez**
### Description

This pull request updates the bdk-ffi crate to use the latest beta version of bdk, specifically version [1.0.0-beta.5](https://github.com/bitcoindevkit/bdk/releases/tag/v1.0.0-beta.5). 

The main changes include:
- Updating the Cargo.toml file.
- Adjusting the RBF functionality to align with the [changes](https://github.com/bitcoindevkit/bdk/pull/1616) in the new BDK version.
  - Removing deprecated methods like enable_rbf() and enable_rbf_with_sequence().
  - Introducing the new [set_exact_sequence](https://docs.rs/bdk_wallet/1.0.0-beta.5/bdk_wallet/struct.TxBuilder.html#method.set_exact_sequence) method to replace the old RBF-related methods
- Updating [errors](https://docs.rs/bdk_wallet/latest/bdk_wallet/error/enum.CreateTxError.html), particularly for the RbfSequenceCsv error variant.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

File: bdk-ffi/Cargo.toml
Comment (thunderbiscuit) [inline bdk-ffi/Cargo.toml:21]: Here I think we should pin the bdk_wallet dependency to beta 5. The default cargo behaviour of updating to the latest patch broke my (and all of our) builds as of yesterday when they released beta 5 which contained a breaking change. This is not a bdk problem but really a cargo problem.

**Review by thunderbiscuit — APPROVED** [review]
Very nice cleanup.

I have one tiny thing I'd like to clean up before merging, but that's it.

Nice work; I like the removal of the explicit rbf setter, and we're probably the first ones to do that by default!

**Review by thunderbiscuit — CHANGES_REQUESTED** [review]
Sorry I chose "Approve" when I should have clicked "Request changes".

File: bdk-ffi/Cargo.toml
Comment (reez) [reply bdk-ffi/Cargo.toml]: I pushed an update to pin, but just double check its what you were asking for and I won't merge or anything until you give it the go ahead that this is good-

**Review by thunderbiscuit — APPROVED** [review]
ACK 6cfcf2312a8878d6c029649185f4c9543fe06f6e. Ready to merge.

**Pull request merged**
